test/IporSwapLogic/SwapLogicCalculateQuasiInterest.t.sol

```diff
@@ -0,0 +1,231 @@
+            9870300000000000000000 * 10, // notional
+            987030000000000000000, // ibtQuantity
+            TestConstants.PERCENTAGE_4_18DEC, // fixedInterestRate
+            TestConstants.TC_LIQUIDATION_DEPOSIT_AMOUNT_18DEC, // liquidationDepositAmount
+            1 // state
+        );
+        // when
+        (uint256 quasiIFixed, uint256 quasiIFloating) = _iporSwapLogic.calculateQuasiInterest(
+            swap, swap.openTimestamp + TestConstants.PERIOD_25_DAYS_IN_SECONDS * 4, 120 * TestConstants.D18
+        );
+        // then
+        assertEq(quasiIFixed, 3146809564800000000000000000000000000000000000000);
+        assertEq(quasiIFloating, 3735237369600000000000000000000000000000000000000);
+    }
+
+    function testShouldCalculateQuasiInterestCase1() public {
+        // given
+        IporTypes.IporSwapMemory memory swap = IporTypes.IporSwapMemory(
+            TestConstants.ZERO, // id
+            _admin, // buyer
+            block.timestamp, // openTimestamp
+            block.timestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, // closeTimestamp
+            TestConstants.ZERO, // idsIndex
+            TestConstants.USD_50_000_18DEC, // collateral
+            9870300000000000000000 * 10, // notional
+            987030000000000000000, // ibtQuantity
+            TestConstants.PERCENTAGE_4_18DEC, // fixedInterestRate
+            TestConstants.TC_LIQUIDATION_DEPOSIT_AMOUNT_18DEC, // liquidationDepositAmount
+            1 // state
+        );
+        // when
+        (uint256 quasiIFixed, uint256 quasiIFloating) = _iporSwapLogic.calculateQuasiInterest(
+            swap, swap.openTimestamp + TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, 1 * TestConstants.D18
+        );
+        // then
+        assertEq(quasiIFixed, 3122249099904000000000000000000000000000000000000);
+        assertEq(quasiIFloating, 31126978080000000000000000000000000000000000000);
+    }
+
+    function testShouldRevertWhenClosingTimestampIsLowerThanOpenTimestamp() public {
+        // given
+        IporTypes.IporSwapMemory memory swap = IporTypes.IporSwapMemory(
+            TestConstants.ZERO, // id
+            _admin, // buyer
+            block.timestamp + TestConstants.PERIOD_50_DAYS_IN_SECONDS, // openTimestamp
+            block.timestamp + TestConstants.PERIOD_75_DAYS_IN_SECONDS, // closeTimestamp
+            TestConstants.ZERO, // idsIndex
+            TestConstants.USD_50_000_18DEC, // collateral
+            9870300000000000000000 * 10, // notional
+            987030000000000000000, // ibtQuantity
+            TestConstants.PERCENTAGE_4_18DEC, // fixedInterestRate
+            TestConstants.TC_LIQUIDATION_DEPOSIT_AMOUNT_18DEC, // liquidationDepositAmount
+            1 // state
+        );
+        // when
+        vm.expectRevert("IPOR_319");
+        _iporSwapLogic.calculateQuasiInterest(
+            swap, swap.openTimestamp - TestConstants.SWAP_DEFAULT_PERIOD_IN_SECONDS, 1 * TestConstants.D18
```

Comment:
Should work swap.openTimestamp - any value.

Reply:
Do I leave as is? Or change with something else?